### PR TITLE
Board configuration for 063-ble

### PR DESCRIPTION
**Pull request by jaenrig-ifx**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

* Moved general configuration to board configuration
* Dimension gc_heap_size as per board
* Dimension flash as per MCU
* Removed SSL and related featuers from 063-BLE to reduce memory

### CODE REVIEW

File: ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h [hunk 1]
```diff
@@ -3,3 +3,10 @@
 #define MICROPY_HW_BOARD_NAME                   "CY8CPROTO-062-4343W"
 
 #define MICROPY_PY_NETWORK_HOSTNAME_DEFAULT     "CY8C-062-4343W"
+
+#define MICROPY_GC_HEAP_SIZE                    (256 * 1024) // 256 KB
+
+#define MICROPY_PY_HASHLIB                      (1)
```
Comment (NikhitaR-IFX) [inline ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h:9]: If I get it right these are defines for mpy extended modules which means they are by default already defined (as 0?) in one of the mpy extmod files. Only if they have to be redefined to 1 , then needs to be included either common to port or board. or?

File: ports/psoc6/boards/CY8CPROTO-063-BLE/mpconfigboard.h [hunk 2]
```diff
@@ -3,3 +3,11 @@
 #define MICROPY_HW_BOARD_NAME                   "CY8CPROTO-063-BLE"
 
 #define MICROPY_PY_NETWORK_HOSTNAME_DEFAULT     "CY8C-063-BLE"
+
+#define MICROPY_GC_HEAP_SIZE                    (64 * 1024) // 64 KB
+
+#define MICROPY_PY_HASHLIB                      (0)
```
Comment (NikhitaR-IFX) [inline ports/psoc6/boards/CY8CPROTO-063-BLE/mpconfigboard.h:9]: Same as comment in previous board mpconfigboard.h

**Review by NikhitaR-IFX — APPROVED** [review]
Rest looks good! 🚀

File: ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h (hunk 1 above)
Comment (jaenrig-ifx) [reply ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h:9]: You are right, I just have removed them from the file.

Comment (jaenrig-ifx) [reply ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h:9]: I have only kept in the old board, and removed it in the 063-BLE.

**Review by IFX-Anusha — APPROVED** [review]

File: ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h (hunk 1 above)
Comment (NikhitaR-IFX) [reply ports/psoc6/boards/CY8CPROTO-062-4343W/mpconfigboard.h:9]: Perfect!

**Pull request merged**